PasteBinService/RegisterService/src/main/java/org/potrohit/springboot/RegisterService/infrastructure/s3/S3AdapterImpl.java

```diff
@@ -0,0 +1,139 @@
+package org.potrohit.springboot.RegisterService.infrastructure.s3;
+
+import org.potrohit.springboot.RegisterService.domain.vao.PasteBinVao;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.*;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+import software.amazon.awssdk.services.s3.presigner.model.GetObjectPresignRequest;
+import software.amazon.awssdk.services.s3.presigner.model.PresignedGetObjectRequest;
+import software.amazon.awssdk.services.s3.presigner.model.PresignedPutObjectRequest;
+import software.amazon.awssdk.services.s3.presigner.model.PutObjectPresignRequest;
+
+import java.net.URL;
+import java.time.Duration;
+import java.util.ArrayList;
+
+@Log4j2
+@Component
+public class S3AdapterImpl {
+
+    @Autowired
+    private final S3Client s3Client = null;
+
+    @Autowired
+    private final S3Presigner s3Presigner = null;
+
+    public S3Details getS3Details(final PasteBinVao pasteBinVao) {
+        return S3Details.builder()
+                .s3Bucket("BUCKET")
+                .build();
+    }
+
+    public String getPreSignedDownloadURL(S3Details s3Data) {
```

Comment:
method should be very short and consize. make it a rule to not exceed more than 10 lines.  You can read clean code where author talks about having just 3-4 lines in a method.